docs/bricks_design.rst

```diff
@@ -0,0 +1,602 @@
+
+Bricks design explained
+=======================
+
+This is a runnable tutorial, which means you can copy its fragments 
+to an IPython notebook and run. The import section follows:
```

Comment:
The problem is they're harder to test automatically (see [this discussion](https://github.com/mila-udem/fuel/issues/195)).